### ISSUE TITLE
Feedback: Fix the website.

### ISSUE THREAD

**Issue by cmpunches** [SILO-GROUP/Pyrois#5]
The DHLP website is horrid.

I'm not a graphic designer.  The use of darkfeed for static content generation is a good security practice, but perhaps a wordpress theme could be retrofitted onto it so that it's a little more palletable to audiences.

**Comment by cmpunches** [comment]
What does "fix the website" really mean though...

**Comment by cmpunches** [comment]
I see 3 aspects of this:

1. **The software used by the website**.  Does it require javascript frameworks that will go out of date over time?  is it static content or dynamic content (which has a weaker security posture and requires more maintenance -- and reduces the lifetime of the site).  Build something strong.
 
2. **The content of the site**. Does it have all the things people would need for the site to be useful?  Documentation, release information and process information etc.  Build something wise.

3. The aesthetic of the site.  Does the aesthetic match with the identity of the brand?  I don't know, I'm not a brand person.  Does it look cool?  Is it easy to navigate?  Are all the things in the right places?  Build something attractive.

**Comment by cmpunches** [comment]
In terms of **content**, to establish a rough baseline, here are some things I think a distro's website _should_ have:

1. A landing page:
- An eye-catching, concise introduction to a new Linux distribution, its features, and the target audience.
- Clear call-to-action buttons for downloading the distribution, exploring documentation, or joining the community.

2. About:
- A detailed explanation of your Linux distribution's goals, philosophy, and unique selling points.
- Information about the team behind the distribution, their expertise, and their motivation for creating it.

3. Download:
- Direct download links for various versions and architectures of your Linux distribution.
- Alternative download methods, such as torrents or mirrors.
System requirements and installation instructions.
Release notes and changelog for each version.

4. Documentation:
- Comprehensive and accessible documentation covering installation, configuration, and troubleshooting.
- Guides for migrating from other distributions or operating systems.
- Tutorials on using your distribution's unique features and tools.
- FAQs addressing common questions and issues.

5.  Community:
- Links to forums, mailing lists, social media, and other platforms where users can connect, share experiences, and provide feedback.
- Information about local user groups or meetups.
- Resources for users who wish to contribute to the project, such as bug reporting, feature requests, or code contributions.

6.  Blog/News:
- Regular updates on the development progress, new features, and important announcements.
- Articles and interviews about the distribution's impact, user stories, and events.
- Security advisories and patches.

7. Support:
- Links to official support channels, such as email or chat.
- A list of commercial support options, when available.
- Links to community-driven support resources, like forums and Q&A sites.

8. Partners and Sponsors:
- Acknowledgment of any partners, sponsors, or organizations that support your distribution.
- Information about how to become a partner or sponsor and the benefits of doing so.

9. Developer Resources:
- Technical documentation and API references for developers who wish to create applications or packages for your distribution.
- Information about the development process, release schedule, and how to contribute to the project.

10.  Legal and Licensing:
- Details about the licenses used for the distribution and its components.
- Terms of use, privacy policy, and any other legal information.

**Comment by cmpunches** [comment]
Here are some established general purpose distro websites to compare to for all 3 legs (strong, wise, pretty):

Canonical's Ubuntu:
https://ubuntu.com/

IBM's RedHat Linux:
https://www.redhat.com/en

Arch Linux:
https://archlinux.org/

CentOS (grain of salt, honestly, failed project): 
https://www.centos.org/

AlmaLinux:
https://almalinux.org/

Slackware (grandfathered in by the old rules):
http://www.slackware.com/

Alpine Linux (don't use except as a minimal baseline- held up by relationships and social issue signals -- they're playing a different game):
https://www.alpinelinux.org/

Skipping Debian's website.  The rules don't apply to them.

Linux Mint:
https://linuxmint.com/

Skipping Fedora because their content pipelines don't scale down.

Manjaro Linux:
https://manjaro.org/

**Comment by cmpunches** [comment]
With the exception of Slackware, every site used for comparison had roughly the same type of content (besides sponsorship) and used a horizontal menu of links at the top of the page for navigation.  

Every site except Slackware used Javascript.

Every distro had a unique logo.

**Comment by cmpunches** [comment]
Ubuntu - landing page starts with a download link, goes to news, some selling points to engineers.  This appears to be development after Ubuntu was established, though, so, what got them here isn't necessarily what they have today.

They have a security section showing DISA/STIG hardening efforts, Common Criteria EAL2: ISO/IEC IS 15408 validation by CSEC and various other security options like FIPS 140-2.

Then a sponsors section.

**Comment by cmpunches** [comment]
Red Hat - landing page starts with some sellling points for strategy for execs to select the OS, goes on to market share stats, emphasizes Fortune 500 client list.  Then some links to various OS-related products, starting with the flagship product (RHEL).  Then openshift, then Ansible Automation Platform (Tower).

Goes on to explore other lines of business, then training resources, CIO-targeted publications, et al.  More formal business accoutrements on the footer.

**Comment by cmpunches** [comment]
Arch Linux - takes a more wiki/sectional based approach.  Landing page starts out with a notice box that:
- explains its the website for arch linux
- explains what arch linux is
- a summary statement about current package set
- a statement about diversity in its community, links to forums and mailing lists to get started, and a reference to the wiki

Then it goes on to news.

Then on the right pane it shows a list of recently updated packages.
From there down on the right pane it covers the same types of content already defined above, and then a donor section below that.

**Comment by cmpunches** [comment]
CentOS - landing page is simple.  Responsive.  Nav bar guides most of the content.


Starts with a header defining the distro, what it's doing:

```
The CentOS Project
Community-driven free software effort focused around the goal of providing a rich base platform for open source communities to build upon.

We offer two Linux variants (link: what's the difference):
```

And then lists its products.

There's then a horizontal pane that includes just icon links to the same items in their community section listed in their nav bar.

There's then a single pane, vertical stack of news and event posts.  Same for community and organizational publications, looks like some financial pubs.

Then a footer much like red hat's.  About page (referenced in navbar), FAQ, SIG/governance shit, CentOS variants/spins, more governance, Code of Conduct.  Then a community panel with the same links in the icon panel previously mentioned (and in the top nav bar -- site looks like its generated or server-side scripting using a framework with menu objects that are reused, like wordpress).

Looks like they use jekyll: https://gitlab.com/areguera/jekyll-theme-centos/

**Comment by cmpunches** [comment]
AlmaLinux -

This impressive distro is more thought out than I'll be able to describe but the first thing you see is a screenshot of their very well polished installer and a project description with a download link, a contribute link, and a migrate link.

Below that is a horizontal plane for community links.

Then a more detailed description and some organizational information, some funding information.

Then a pane for sponsors.

Then a pane with 3 selling points.

Then a pane for orgs that provide commercial support.  (Interesting, wouldn't they be competing with these orgs?)

Then a collapsible Q&A section.

Then a pane for mailing lists and links for staying up to date.

Then a section for references to AlmaLinux in the press.

Then a footer section. Resources (wiki, github, bug tracker, repos, downloads, membership, pilot program, mailing lists, status page), Community (forums, reddit, twitter, social shit), Legal (legal notice, privacy policy, terms of service, licensing policy, trademark usage, foundation bylaws, membership committee).

These guys seem really good at this. Not too bloated either.

Site aesthetic is a hugo theme.

**Comment by cmpunches** [comment]
Slackware - 

Probably has different rules to meet than the others.  No sponsors.  Broken html.  No javascript.  The distro serves a kind of nostalgic community value.  

Uses mailing list subscription.  Majordomo (perl) behind postfix it looks like.

Everything's GPG signed in notices.

- Security advisory (mailing list is published)
- FAQs
- Book (LFS-style documentation project on a different domain)
- General Info (more FAQ)
- Download page
- Changelogs
- Package downloads (no network based package management or dependency management)
- *marketing material*
- spins to ARM architecture
- other related sites
- support - list of companies that do commercial support of slackware
- contact deets - various email addresses, a mirroring program
- mailing lists
- about page

There's a nostalgia component, and this should not be used as a guide.

**Comment by cmpunches** [comment]
All distros observed so far have a short, one-word name or a juxtoposition of two words.*

**Comment by cmpunches** [comment]
Alpine Linux

Everything in the navbar except the landing page, which is just a slogan, a one sentence product description, and two news columns -- one for product lifecycle, and one for package changes.

In the navbar, is:
- about page
- downloads page
- release branches page (just a list and nav of all their git branches)
- community (where to go to get support, mailing lists, a link to a wiki, and some social media profile links)
- sponsors page
- docs
- wiki
- git repo
- issues (direct on their gitlab instance)
- packages (a searchable list of packages)
- mirrors page (list of all mirrors hosting their content)
- security page (separate issue trackers for security issues)

short, sweet, simple

**Comment by cmpunches** [comment]
Linux Mint

Typical navbar and landing as the others.

Notable feature is they have both sponsors and donors.

Sponsors have several tiers: Gold, Silver, Bronze, Community (1k/mo, 500/mo, 100/mo, 20/mo).  Higher tiers get more prominent placement on the sponsors list, as well as placement on different parts of the site in a sponsors banner, in addition to mentions in announcements.

Donors are more for smaller donations and non-recurring donations, but it does offer a monthly patreon.

They take donations in USD/EUR, as well as several crypto currencies (BTC, ETH, LTC, BCH, XMR).

Then they listed:
- latest donors
- top 10 countries
- top 100 donors.

They do seem to promote their forum a little more.  Self-hosted, phpbb.

geared for community and general purpose desktop.

**Comment by cmpunches** [comment]
Manjaro

Nothing not already covered.  Download link is obvious on first view.  Supporters section is easy to find.  Lots of scrolling.

**Comment by cmpunches** [comment]
Aspect 1:

Static site generation on a Fedora or alma stack.  
Any dynamic content such as news will be from the feed aggregator, and also is static content generated for that specific section.  Every piece will be an independent piece sharing a common theme and no server-side scripting will be exposed.  Themes will be retrofit onto the generator.

**Comment by cmpunches** [comment]
Navbar Content will reflect site structure:

- home/landing
- about
  - distribution
  - team
- download
- documentation
  - installation
  - configuration
  - troubleshooting
  - migration guides
  - tutorials/wiki
  - FAQs
- community
  - platforms where users can connect, share experience, provide feedback
  - contributors (link to top-level contributor page)
- Blog/News
  - regular updates/progress/announcements (high level)
- Support
  - Commercial Support
  - Free Community Support
- Partners / Sponsors
  - sponsors page
  - how to become a partner or sponsor
  - benefits of sponsorship
- developer resource page
  - how to contribute
  - development process and pipelines
  - bug reporting / feature requests links
- legal
  - ownership
  - licensing
  - terms of use
  - privacy policy

need a 2 layer deep navbar

**Comment by cmpunches** [comment]
observation:

while alma has the content and organization down, they have some continuity issues with the website as a result of organic growth on top of the hugo theme, where different parts are managed by different systems.

**Comment by cmpunches** [comment]
the hugo test drive was helpful. 

hugo is not desirable for this. 

build a static site generator.
